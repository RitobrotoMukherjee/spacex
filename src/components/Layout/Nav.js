import '../../css/Navigation.css';
import { NavLink, Link } from 'react-router-dom';

const Navigation = () => (
  <header>
    <nav className="nav">
      <div className="logo Flex-1">
        <Link to="/">
          <img src={`${process.env.PUBLIC_URL}/logo.png`} alt="Spacex" />
        </Link>
        <h2>Space Travellers Hub</h2>
      </div>
      <ul className="navLinks Flex-1">
        <li className="linkItems">
          <NavLink
            className={({ isActive }) => (isActive ? 'activeClassName' : '')}
            to="rockets"
          >
            Rockets
          </NavLink>
        </li>
        <li className="linkItems mission-link">
          <NavLink
            className={({ isActive }) => (isActive ? 'activeClassName' : '')}
            to="missions"
          >
            Missions
          </NavLink>
        </li>
        <li className="linkItems">
          <NavLink
            className={({ isActive }) => (isActive ? 'activeClassName' : '')}
            to="my-profile"
          >
            My Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  </header>
);

export default Navigation;
