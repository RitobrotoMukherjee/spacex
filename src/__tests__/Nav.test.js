import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Nav from '../components/Layout/Nav';

describe('Testing the Nav', () => {
  it('Nav is rendered', () => {
    const nav = renderer
      .create(<BrowserRouter><Nav /></BrowserRouter>)
      .toJSON();
    expect(nav).toMatchSnapshot();
  });
});
