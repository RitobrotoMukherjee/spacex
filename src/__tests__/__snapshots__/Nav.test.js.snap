// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Nav Nav is rendered 1`] = `
<header>
  <nav
    className="nav"
  >
    <div
      className="logo Flex-1"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Spacex"
          src="/logo.png"
        />
      </a>
      <h2>
        Space Travellers Hub
      </h2>
    </div>
    <ul
      className="navLinks Flex-1"
    >
      <li
        className="linkItems"
      >
        <a
          className=""
          href="/rockets"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="linkItems mission-link"
      >
        <a
          className=""
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="linkItems"
      >
        <a
          className=""
          href="/my-profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
